python/MetrologyData.py

```diff
@@ -69,10 +71,93 @@ def xyzPlane_fit(self, nsigma=3, p0=(0, 0, 0)):
         dz = xyz_plane(positions, *pars) - self.z
         mean, stdev = np.mean(dz), np.std(dz)
 
-        # Refit the reference data within nsigma*stdev of the mean.
+        # Refit iteratively until the standard deviation of residuals does not
+        # change
+        stdev_last = -1
+        stdev_new = stdev
         index = np.where((dz > mean-nsigma*stdev) & (dz < mean+nsigma*stdev))
-        pars, _ = scipy.optimize.curve_fit(xyz_plane, positions[index],
-                                           self.z[index], p0=pars)
+        while stdev_new != stdev_last:
+            stdev_last = stdev_new
+            # Refit the reference data within nsigma*stdev of the mean.
+            pars, _ = scipy.optimize.curve_fit(xyz_plane, positions[index],
+                                               self.z[index], p0=pars)
+            dz = xyz_plane(positions, *pars) - self.z
+            mean, stdev_new = np.mean(dz[index]), np.std(dz[index])
+            index = np.where((dz > mean-nsigma*stdev_new) &
+                             (dz < mean+nsigma*stdev_new))
+
+        # Make the standard deviation and mean of the filtered residuals
+        # available, along with the fit parameters
+        self.stdev_filt = stdev_new
+        self.mean_filt = mean
+        self.pars = pars
+
+        print(pars)
+        # For this set of 'good' points, model the expected distribution of
+        # residuals based on fitting planes to the individual CCD surfaces.
+        # This involves stepping through the CCDs (S00 -> S22), selecting the
+        # data points 'owned' by a given sensor, fitting a plane to them,
+        # and evaluating the plane at each of the data points owned by the CCD.
+        # An array with the model residuals with respect to the overall
+        # best-fitting plane is accumulated
+        x = self.x
+        y = self.y
+        # Construct an 'ideal' list of x, y grid points spanning the sensor, so that
+        # positions excluded from the fits as outliers are not excluded from the
+        # evaluation of the fitted sensor surface height
+        x1 = np.arange(np.min(x), np.max(x)+4, 4)
+        y1 = np.arange(np.min(y), np.max(y)+4, 4)
+        xmodel = []
+        ymodel = []
+        for xi in x1:
+            for yi in y1:
+                xmodel.append(xi)
+                ymodel.append(yi)
+
+        xmodel = np.asarray(xmodel)
+        ymodel = np.asarray(ymodel)
+
+        for ccd in range(0, 9):
```

Comment:
I should have scrutinized these changes more carefully in the initial review.   This code needs to be used for single sensor metrology.  That fact that it is looping over `ccd`s (apparently in a raft) is a clear indication that these changes won't work in SR-MET-05, for example.  We're going to have to back out these changes.